src/Compatibility/Microsoft.DotNet.ApiCompat.Task/ValidateAssembliesTask.cs

```diff
@@ -0,0 +1,126 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.IO;
+using Microsoft.Build.Framework;
+using Microsoft.NET.Build.Tasks;
+using Microsoft.DotNet.ApiCompatibility.Logging;
+
+namespace Microsoft.DotNet.ApiCompat.Task
+{
+    public class ValidateAssembliesTask : TaskBase
+    {
+        [Required]
+        public string? RoslynAssembliesPath { get; set; }
+
+        [Required]
+        public string[]? LeftAssemblies { get; set; }
+
+        [Required]
+        public string[]? RightAssemblies { get; set; }
+
+        public bool GenerateCompatibilitySuppressionFile { get; set; }
+
+        public string? CompatibilitySuppressionFilePath { get; set; }
+
+        public string? NoWarn { get; set; }
+
+        public bool EnableStrictMode { get; set; }
+
+        public string[]? LeftAssembliesReferences { get; set; }
+
+        public string[]? RightAssembliesReferences { get; set; }
+
+        public string? SemaphoreFile { get; set; }
+
+        public bool CreateWorkItemPerAssembly { get; set; }
+
+        public ITaskItem[]? LeftAssembliesTransformationPattern { get; set; }
+
+        public ITaskItem[]? RightAssembliesTransformationPattern { get; set; }
+
+        public override bool Execute()
+        {
+            RoslynResolver.Register(RoslynAssembliesPath);
+
+            try
+            {
+                return base.Execute();
+            }
+            finally
+            {
+                RoslynResolver.Unregister();
+            }
+        }
+
+        protected override void ExecuteCore()
+        {
+            Func<ISuppressionEngine, MSBuildCompatibilityLogger> logFactory = (suppressionEngine) => new(Log, suppressionEngine);
+            ValidateAssemblies.Run(logFactory,
+                GenerateCompatibilitySuppressionFile,
+                CompatibilitySuppressionFilePath,
+                NoWarn,
+                LeftAssemblies!,
```

Comment:
This is a parameter that gets passed in, if the calling API doesn't take null then should we be checking for null before calling Run()?

Reply:
Oh, I think I recall we discussed this in the past and you said MSBuild would ensure it is not null.